src/Appwrite/Platform/Workers/Messaging.php

```diff
@@ -198,7 +218,10 @@ private function processMessage(Database $dbForProject, Document $message): void
 
                                 // Deleting push targets when token has expired.
                                 if ($detail['error'] === 'Expired device token.') {
-                                    $target = $dbForProject->findOne('targets', [Query::equal('identifier', [$detail['recipient']])]);
+                                    $target = $dbForProject->findOne('targets', [
+                                        Query::equal('identifier', [$detail['recipient']])
+                                    ]);
+
                                     if ($target instanceof Document && !$target->isEmpty()) {
                                         $dbForProject->deleteDocument('targets', $target->getId());
                                     }
```

Comment:
Looks like the conversation about this is still open: https://github.com/appwrite/appwrite/pull/7394#discussion_r1446129611

Reply:
So this should be updated to queue a delete, right?

Reply:
Handled in [this PR](https://github.com/appwrite/appwrite/pull/7460) which requires this PR first